datahub-frontend/app/client/AuthServiceClient.java

```diff
@@ -88,6 +96,150 @@ public String generateSessionTokenForUser(@Nonnull final String userId) {
+   * Call the Auth Service to reset credentials for a native DataHub user.
+   */
+  @Nonnull
+  public boolean resetNativeUserCredentials(@Nonnull final String email, @Nonnull final String password,
+      @Nonnull final String resetToken) {
+    Objects.requireNonNull(email, "email must not be null");
+    Objects.requireNonNull(password, "password must not be null");
+    Objects.requireNonNull(resetToken, "reset token must not be null");
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    try {
+
+      final String protocol = this.metadataServiceUseSsl ? "https" : "http";
+      final HttpPost request = new HttpPost(
+          String.format("%s://%s:%s/%s", protocol, this.metadataServiceHost, this.metadataServicePort,
+              RESET_NATIVE_USER_CREDENTIALS_ENDPOINT));
+
+      // Build JSON request to verify credentials for a native user.
+      String json =
+          String.format("{ \"%s\":\"%s\", \"%s\":\"%s\", \"%s\":\"%s\" }", EMAIL_FIELD, email, PASSWORD_FIELD, password,
+              RESET_TOKEN_FIELD, resetToken);
+      request.setEntity(new StringEntity(json));
+
+      // Add authorization header with DataHub frontend system id and secret.
+      request.addHeader(Http.HeaderNames.AUTHORIZATION, this.systemAuthentication.getCredentials());
+
+      CloseableHttpResponse response = httpClient.execute(request);
+      final HttpEntity entity = response.getEntity();
+      if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK && entity != null) {
+        // Successfully generated a token for the User
+        final String jsonStr = EntityUtils.toString(entity);
+        return getAreNativeUserCredentialsResetFromJson(jsonStr);
+      } else {
+        throw new RuntimeException(
+            String.format("Bad response from the Metadata Service: %s %s", response.getStatusLine().toString(),
+                response.getEntity().toString()));
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to reset credentials for user", e);
+    } finally {
+      try {
+        httpClient.close();
+      } catch (Exception e) {
+        log.warn("Failed to close http client", e);
+      }
+    }
+  }
+
+  /**
+   * Call the Auth Service to verify the credentials for a native Datahub user.
+   */
+  @Nonnull
+  public boolean verifyNativeUserCredentials(@Nonnull final String email, @Nonnull final String password) {
+    Objects.requireNonNull(email, "email must not be null");
+    Objects.requireNonNull(password, "password must not be null");
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    try {
+
+      final String protocol = this.metadataServiceUseSsl ? "https" : "http";
+      final HttpPost request = new HttpPost(
+          String.format("%s://%s:%s/%s", protocol, this.metadataServiceHost, this.metadataServicePort,
+              VERIFY_NATIVE_USER_CREDENTIALS_ENDPOINT));
+
+      // Build JSON request to verify credentials for a native user.
```

Comment:
We should probably use the real JSON builder libs now that we are building requests to backend APIs more often.

Reply:
Not blocking, but a good followup / backlog item